packages/generator/tests/acceptance/test_pin_aware_measurement_client.py

```diff
@@ -0,0 +1,96 @@
+import importlib.util
+import pathlib
+from types import ModuleType
+from typing import Generator
+
+import grpc
+import pytest
+from ni_measurement_plugin_sdk_service.measurement.service import MeasurementService
+
+from ni_measurement_plugin_sdk_generator.client import create_client
+from tests.utilities.discovery_service_process import DiscoveryServiceProcess
+from tests.utilities.measurements import pin_aware_measurement
+
+
+def test___measurement_plugin_client___measure_with_pin_map_registration___returns_output(
+    measurement_plugin_client_module: ModuleType,
+    pin_map_directory: pathlib.Path,
+) -> None:
+    pin_map_path = str(pin_map_directory / "1Smu1ChannelGroup1Pin1Site.pinmap")
+    output_type = getattr(measurement_plugin_client_module, "Output")
+    expected_output = output_type(
+        pin_map_id=pin_map_path,
+        sites=[0],
+        session_names=["DCPower1/0"],
+        resource_names=["DCPower1/0"],
+        channel_lists=["DCPower1/0"],
+    )
+    test_measurement_client_type = getattr(measurement_plugin_client_module, "TestMeasurement")
+    measurement_plugin_client = test_measurement_client_type()
+
+    measurement_plugin_client.register_pin_map(pin_map_path)
+    output = measurement_plugin_client.measure()
+
+    assert output == expected_output
+
+
+def test___measurement_plugin_client___measure_without_pin_map_registration___raises_no_sessions_error(
+    measurement_plugin_client_module: ModuleType,
+) -> None:
+    test_measurement_client_type = getattr(measurement_plugin_client_module, "TestMeasurement")
+    measurement_plugin_client = test_measurement_client_type()
+
+    with pytest.raises(grpc.RpcError) as exc_info:
+        _ = measurement_plugin_client.measure()
+
+    assert exc_info.value.code() == grpc.StatusCode.UNKNOWN
+    assert "No sessions reserved." in (exc_info.value.details() or "")
+
+
+@pytest.fixture(scope="module")
```

Comment:
altering the fixture scope (from default) is a code-smell.

If you need to do this, please add a comment (here and on the others) *why*
Are these expensive to setup? (while still be safe to re-use?)

Reply:
The scope of this fixture is changed to `module` as this fixture creates a client for the measurement service. Since this setup is expensive, we made this fixture to be reusable throughout the module. This method of caching the fixtures is used in many tests. So, please let us know if you want to add an explicit comment only on this.

Reply:
please add this as a comment 
```suggestion
# Fixture is setup to be re-used across tests due to expense to setup.
@pytest.fixture(scope="module")
```
(or something like that)

Reply:
@mshafer-NI IMO, scope=module is fine for fixtures in a test module, and I would rather not write comments stating the obvious.

What has been a problem lately is moving fixtures from a test module to conftest.py without changing the scope. I see no reason for conftest.py to ever use scope=module. If it's expensive enough to cache, then it's expensive enough to cache across modules.